services/checks/checks.go

```diff
@@ -474,49 +477,82 @@ func (s *Service) minPMMAgentVersion(t check.Type) *version.Parsed {
-func (s *Service) executeChecks(ctx context.Context, interval check.Interval) error {
-	s.l.Infof("Executing %s checks...", interval)
+// filterChecks filters checks by several parameters. If group specified then only matched checks will be returned,
+// empty group means `any interval`. If enable slice is specified then only matched checks will be returned, empty
+// enable slice means `all enabled`. Checks specified in disabled slice are skipped, empty `disabled` slice means
+// `nothing disabled`.
+func (s *Service) filterChecks(checks []check.Check, group check.Interval, disable, enable []string) []check.Check {
+	var res []check.Check
+	disableMap := make(map[string]struct{}, len(disable))
+	for _, e := range disable {
+		disableMap[e] = struct{}{}
+	}
+
+	enableMap := make(map[string]struct{}, len(enable))
+	for _, e := range enable {
+		enableMap[e] = struct{}{}
+	}
+
+	for _, c := range checks {
+		// If empty group passed, which means `any group`
+		// or check has required interval
+		// or check has empty interval and required interval is `standard`.
+		if group == "" || c.Interval == group || (group == check.Standard && c.Interval == "") {
+			// If check enabled explicitly or all checks enabled by passing empty `enable` slice.
+			if _, ok := enableMap[c.Name]; ok || len(enableMap) == 0 {
+				// Filter disabled checks.
+				if _, ok := disableMap[c.Name]; ok {
+					s.l.Warnf("Check %s is disabled, skipping it.", c.Name)
+					continue
+				}
+
+				res = append(res, c)
+			}
+		}
+	}
 
+	return res
+}
+
+// executeChecks runs checks for all reachable services. If group specified only checks from that group will be executed.
+// If checkNames specified then only matched checks will be executed.
+func (s *Service) executeChecks(ctx context.Context, group check.Interval, checkNames []string) error {
 	disabledChecks, err := s.GetDisabledChecks()
 	if err != nil {
 		return errors.WithStack(err)
 	}
 
 	var checkResults []sttCheckResult
 
-	mySQLCheckResults := s.executeMySQLChecks(ctx, interval, disabledChecks)
+	mySQLChecks := s.filterChecks(s.getMySQLChecks(), group, disabledChecks, checkNames)
+	mySQLCheckResults := s.executeMySQLChecks(ctx, mySQLChecks)
 	checkResults = append(checkResults, mySQLCheckResults...)
 
-	postgreSQLCheckResults := s.executePostgreSQLChecks(ctx, interval, disabledChecks)
+	postgreSQLChecks := s.filterChecks(s.getPostgreSQLChecks(), group, disabledChecks, checkNames)
+	postgreSQLCheckResults := s.executePostgreSQLChecks(ctx, postgreSQLChecks)
 	checkResults = append(checkResults, postgreSQLCheckResults...)
 
-	mongoDBCheckResults := s.executeMongoDBChecks(ctx, interval, disabledChecks)
+	mongoDBChecks := s.filterChecks(s.getMongoDBChecks(), group, disabledChecks, checkNames)
+	mongoDBCheckResults := s.executeMongoDBChecks(ctx, mongoDBChecks)
 	checkResults = append(checkResults, mongoDBCheckResults...)
 
-	s.alertsRegistry.set(interval, checkResults)
+	if len(checkNames) != 0 {
```

Comment:
Is it possible that we run the same checks for different groups?

Reply:
Nope, a single check can be a member of a single group.